### PR TITLE
Fix Ordinal Encoder

### PR DESCRIPTION
**Pull request by samborba**
Classification and Regression Models:
- Fix Ordinal Enconder: The OrdinalEncoder method was not being applied correctly using the numerical_indexes variable, as ColumnTransformer was modifying the input indexes. To fix this problem, a new mask was used to keep track of features of the categorical type.

-  Fix ColumnTransformer empty list handling: If the input of any transformer of the ColumnTransformer method received an empty list, it generated an error because it will have no action to be done. To resolve this behavior, it is necessary to check before passing as an argument to the method. A new cell ("Feature configuration") with this check was made.

-  Add encode target label: Some notebooks were without this treatment, which could cause some type of error in the future.

Attribute engineering:
- Small changes in Imputer

### CODE REVIEW

**Comment by samborba** [comment]
@fberanizo como conversamos na sexta: quando algum step do ColumnTransformer recebe um array com tudo falso (e.g. [False, False, False]) um erro é gerado falando que precisa de pelo menos uma coluna pra tomar alguma ação e não deixa continuar.
Tem uma [issue](https://github.com/scikit-learn/scikit-learn/issues/12071) no repositório do Skelearn que reclama desse comportamento. Teve um PR pra resolver isso, mas só se aplica pra quando o array é vazio e no nosso caso o array é todo False, o que era pra ser o mesmo comportamento... eu fiz uma pergunta lá pra ver se é isso mesmo, mas enquanto isso eu fiz um workaround pra tratar isso (que está na célula "Feature configuration").

**Comment by fberanizo** [comment]
@samborba parece que funciona com array vazio mesmo. Neste caso, você pode usar o `np.where` e a própria biblioteca do sklearn. Fiz uns testes aqui e parece que deu certo:

Iris Dataset:
Screenshot_2020-05-24 JupyterLab(2)


Titanic Dataset:
Screenshot_2020-05-24 JupyterLab

**Comment by samborba** [comment]
@fberanizo quando eu uso o **OrdinalEncoder** do sklearn, gera erro ao fazer predict. Ele consegue fazer o fit, mas não consegue fazer o predict, já com a biblioteca **category_encoders** dá certo em todas as etapas. 

Print do erro ao fazer predict: 
mlp-classifier-predict-error

**Comment by samborba** [comment]
@fberanizo em relação aos notebooks que fazem **save_dataset**, estou tratando com estas formas:
- Para o Normalizer e Robust Scaler: no remainder do **ColumnTransformer** eu deixo como "drop" e o resultado do **pipepline.fit_transform** eu atribuo para o X[:, numerical_indexes], já que os modelos só trabalham com numerical e não dão drop em features.

- Para o Imputer: se o strategy for "mean" ou "median" eu faço a mesma ação de cima (do Normalizer e Robust Scaler) e se for "most_frequent" eu atribuo o resultado do **pipepline.fit_transform** eu atribuo ao X pois ele aplica pros dois tipos de features (categorical e numerical).

- Para o Variance Threshold: eu jogo o resultado do **pipeline.fit_transform** (passando como argumento os numericals) para um variável X_thr e depois disso eu concateno ao X junto com os categoricals e reorganizo as colunas de acordo com o resultado do pipeline.
